pyblazing/pyblazing/apiv2/context.py

```diff
@@ -681,36 +681,57 @@ def adjust_due_missing_rowgroups(metadata, files):
 def distributed_initialize_server_directory(client, dir_path):
 
     # lets make host_list which is a list of all the unique hosts.
-    # This way we do the logging folder creation only once per host (server)
+    # this way we do the logging folder creation only once per host (server)
+    # or for each worker whenever run from different directories
     all_items = client.scheduler_info()["workers"].items()
-    host_list = list(set([value["host"] for key, value in all_items]))
-    initialized = {}
-    for host in host_list:
-        initialized[host] = False
 
     dask_futures = []
-    for worker, worker_info in client.scheduler_info()["workers"].items():
-        if not initialized[worker_info["host"]]:
-            dask_futures.append(
-                client.submit(initialize_server_directory, dir_path, workers=[worker])
-            )
-            initialized[worker_info["host"]] = True
+    i = 0  # this extra index is to avoid scheduler optimization
```

Comment:
A better way of doing this (that we learned recently) is to do like this:
```
for worker, worker_info in all_items:
            dask_futures.append(	        
                client.submit(initialize_server_directory, dir_path, workers=[worker], pure=False)	            
            )	        
```
Using the parameter `pure=False` makes it so that you dont have to pass it that `i`

Reply:
Fixed, works perfectly, thanks!